### PR TITLE
AM-766 SCIM group endpoint filter

### PR DESCRIPTION
**Pull request by ashraf706**
## How To Test:

The easiest way would be to use existing Postman SCIM collection.

1. Follow the steps and create a couple users and groups.
2. Add users to a group [we have Patch group - add operation]
3. In the Postman `List` folder we have `List Groups` which lists all the groups.
4. Add a new GET request such as `{{gateway_url}}/{{domainHrid}}/scim/Groups?filter=displayName eq "Tour Guides"` should return only the mentioned group.

Please check with any JDBC as well if possible.

### CODE REVIEW

File: gravitee-am-repository/gravitee-am-repository-jdbc/src/main/java/io/gravitee/am/repository/jdbc/common/dialect/AbstractDialectHelper.java [hunk 1]
```diff
@@ -172,6 +172,8 @@ protected final String convertFieldName(FilterCriteria criteria) {
                 return "enabled";
             case "emails.value":
                 return "email";
+            case "displayName":
+                return "name";
```
Comment (leleueri) [inline gravitee-am-repository/gravitee-am-repository-jdbc/src/main/java/io/gravitee/am/repository/jdbc/common/dialect/AbstractDialectHelper.java:176]: **issue**: User has name and displayName, with this addition you will break the user filtering IMO. Can you check that point please ?

Comment (ashraf706) [reply gravitee-am-repository/gravitee-am-repository-jdbc/src/main/java/io/gravitee/am/repository/jdbc/common/dialect/AbstractDialectHelper.java]: It breaks, had to create a separate method to convert Groups name

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-scim/src/main/java/io/gravitee/am/gateway/handler/scim/service/impl/GroupServiceImpl.java [hunk 2]
```diff
@@ -68,10 +71,14 @@ public class GroupServiceImpl implements GroupService {
...
 
     @Override
-    public Single<ListResponse<Group>> list(int page, int size, String baseUrl) {
+    public Single<ListResponse<Group>> list(Filter filter, int page, int size, String baseUrl) {
         LOGGER.debug("Find groups by domain : {}", domain.getId());
 
-        return groupRepository.findAll(ReferenceType.DOMAIN, domain.getId(), page, size)
+        Single<Page<io.gravitee.am.model.Group>> groupResult = filter != null ?
+                groupRepository.search(ReferenceType.DOMAIN, domain.getId(), FilterCriteria.convert(filter), page, size) :
+                groupRepository.findAll(ReferenceType.DOMAIN, domain.getId(), page, size);
```
[the remark this answers is not in the document]
Comment (ashraf706) [reply gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-scim/src/main/java/io/gravitee/am/gateway/handler/scim/service/impl/GroupServiceImpl.java:79]: test added

**Review by leleueri — APPROVED** [review]
**TODO:** looks ok now, can you please check the Sonar issue. Looks like we have potential NullPointerException (not linked to your change but could b enice to check and fix if possible)

**Pull request merged**